libs/core/threading_base/src/set_thread_state.cpp

```diff
@@ -134,8 +134,8 @@ namespace hpx { namespace threads { namespace detail {
                         get_thread_state_name(new_state));
 
                     thread_init_data data(
-                        util::bind(&set_active_state, thrd, new_state,
-                            new_state_ex, priority, previous_state),
+                        util::bind(&set_active_state, thread_id_ref_type(thrd),
+                            new_state, new_state_ex, priority, previous_state),
```

Comment:
Is there a difference here compared to the old code? This is explicitly making a copy, whereas the old code would implicitly make a copy, but the result is the same, right?

Reply:
The old code did make a copy of a non-reference counted thread-id.

Reply:
With `thrd` a `thread_id_ref_type`, couldn't you just pass `thrd` to `bind` and it would behave exactly the same? Sorry if I'm being thick. I just want to understand if I'm missing something and there's a functional difference between the two.

Reply:
AFAICT, `thrd` is a non-reference counted thread-id (see https://github.com/STEllAR-GROUP/hpx/pull/5546/files/70bbe3eab3988c43977b636e4cb329f3c4ee645a#diff-93346ad9af6b7251c1205b63dcfe560916d8df9b783850442dc6bedecd52b71cR70)

Reply:
Sorry for the noise. You're absolutely right. I just don't know how to read the diff. It makes absolute sense now.